[8403008] Add client connection related edgeHub metrics (#4060)
Azure/iotedge: doc/BuiltInMetrics.md

| `edgehub_messages_unack_total`                              |  `reason` (storage_failure)<br> `from` (Message source)<br> `from_route_output` (Output that sent the message)<br>	| Total number of messages unack because storage failure			| counter |
| `edgehub_offline_count_total`                               |  `id` (Module ID)<br>                                                      									| Total number of times edgeHub went offline								| counter | 
| `edgehub_offline_duration_seconds`                          |  `id` (Module ID)<br> `quantile`(Percentile [50, 90, 95, 99, 99.9, 99.99]) 									| P50, P90, P95, P99, P99.9 and P99.99 time edge hub was offline. Values may be reported as `NaN`  if no new measurements are reported for a  certain period of time (currently 10 minutes). As this is `summary` type, corresponding `_count` and `_sum` counters will be emitted.              	| summary |
| `edgehub_operation_retry_total`      						  |  `id` (Module ID)<br>`operation` (Operation name)                                                        	| Total number of times edgeHub operations were retried 					| counter | 

> | `edgehub_client_connect_failed_total`                       | `id` (Device/Module ID)<br> `reason` (not authenticated)<br> 												| Total number of times individual client failed to connect to edgeHub		| counter |                                                              

Reviewer: ```suggestion
| `edgehub_client_connect_failed_total`                       | `id` (Device/Module ID)<br> `reason` (not authenticated)<br> 												| Total number of times each client failed to connect to edgeHub		| counter |                                                              
```

> | `edgehub_connected_clients`                              	  | 															 												| Current number of connected clients to edge hub        					| gauge |                                                            

Reviewer: ```suggestion
| `edgehub_connected_clients`                              	  | 															 												| Current number of clients connected to edgeHub        					| gauge |                                                            
```

> | `edgehub_client_connect_success_total`                      | `id` (Device/Module ID)<br>																					| Total number of times each client successfully connected to edgeHub   | counter |                                                                                                                                                                    | counter |                                                              

Reviewer 2: 1. With the current implementation, there is no distinction between have the same module connected & reconnected multiple times vs having multiple modules connect at the same time. I'm not sure what the customer use case is, but I'm having a hard time interpreting the meaning of this metric.
2. This metric only accurate per edgeHub restart cycle.

Author: 1. There is `id` in the metrics which will tell which device/module it is. The counter will group based on the `id`.
2. Ideally metrics should be collected periodically. In user scenario if customer  would like to monitor edge hub, the metrics should be feed to Azure monitoring or tracking or ELK search.  Restart causes metrics data rest shouldn't be considered in the implementation unless we'd like to persist it locally.

> | `edgehub_client_connect_success_total`                      | `id` (Device/Module ID)<br>																					| Total number of times individual client successfully connect to edgeHub   | counter |                                                                                                                                                                    | counter |                                                              

Reviewer: ```suggestion
| `edgehub_client_connect_success_total`                      | `id` (Device/Module ID)<br>																					| Total number of times each client successfully connected to edgeHub   | counter |                                                                                                                                                                    | counter |                                                              
```

> | `edgehub_client_disconnect_total`                      	  | `id` (Device/Module ID)<br>																					| Total number of times individual client disconnected from edgeHub   		| counter |

Reviewer: ```suggestion
| `edgehub_client_disconnect_total`                      	  | `id` (Device/Module ID)<br>																					| Total number of times each client disconnected from edgeHub   		| counter |
```
